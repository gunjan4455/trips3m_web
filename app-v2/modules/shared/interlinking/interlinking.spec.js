import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import Interlinking from './index.js';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  footerLinks: []
};

const shallowWrapperInterlinking = (props) => {
  return shallow(<Interlinking {...props} />);
};

describe('Interlinking', () => {

  let props = { ...DEFAULT_PROPS };

  it('should render correctly', () => {
    const wrapper = shallowWrapperInterlinking(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when footerLinks has data', () => {
    props = {
      footerLinks: [
        { name: 'Property Types in Bali',
          footerLinks: [
            {name: 'Hotels in Bali', url: ''},
            { name: 'Guest Houses in Bali',  url: ''}
          ]
        }
      ]
    };
    const wrapper = shallowWrapperInterlinking(props);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h3')).toHaveLength(1);
    expect(wrapper.find('h3').text()).toEqual('Property Types in Bali');
    expect(wrapper.contains('.overflowXscroll'));
  });

});