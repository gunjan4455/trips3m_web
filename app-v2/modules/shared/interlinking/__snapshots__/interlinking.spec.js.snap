// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interlinking should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InterLinking
    footerLinks={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Interlinking should render correctly when footerLinks has data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InterLinking
    footerLinks={
      Array [
        Object {
          "footerLinks": Array [
            Object {
              "name": "Hotels in Bali",
              "url": "",
            },
            Object {
              "name": "Guest Houses in Bali",
              "url": "",
            },
          ],
          "name": "Property Types in Bali",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="p15 sbcw mb8"
        >
          <h3
            className="f16 fw9 pfc3 mb15"
          >
            Property Types in Bali
          </h3>
          <div
            className="overflowXscroll swiper-full-width"
          >
            <div
              className="flex mb8"
            >
              <RenderLinks
                name="Hotels in Bali"
                url=""
              />
              <RenderLinks
                name="Guest Houses in Bali"
                url=""
              />
            </div>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3
              className="f16 fw9 pfc3 mb15"
            >
              Property Types in Bali
            </h3>,
            <div
              className="overflowXscroll swiper-full-width"
            >
              <div
                className="flex mb8"
              >
                <RenderLinks
                  name="Hotels in Bali"
                  url=""
                />
                <RenderLinks
                  name="Guest Houses in Bali"
                  url=""
                />
              </div>
            </div>,
          ],
          "className": "p15 sbcw mb8",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Property Types in Bali",
              "className": "f16 fw9 pfc3 mb15",
            },
            "ref": null,
            "rendered": "Property Types in Bali",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="flex mb8"
                >
                  <RenderLinks
                    name="Hotels in Bali"
                    url=""
                  />
                  <RenderLinks
                    name="Guest Houses in Bali"
                    url=""
                  />
                </div>,
              ],
              "className": "overflowXscroll swiper-full-width",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <RenderLinks
                      name="Hotels in Bali"
                      url=""
                    />,
                    <RenderLinks
                      name="Guest Houses in Bali"
                      url=""
                    />,
                  ],
                  "className": "flex mb8",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "name": "Hotels in Bali",
                      "url": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "name": "Guest Houses in Bali",
                      "url": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="p15 sbcw mb8"
          >
            <h3
              className="f16 fw9 pfc3 mb15"
            >
              Property Types in Bali
            </h3>
            <div
              className="overflowXscroll swiper-full-width"
            >
              <div
                className="flex mb8"
              >
                <RenderLinks
                  name="Hotels in Bali"
                  url=""
                />
                <RenderLinks
                  name="Guest Houses in Bali"
                  url=""
                />
              </div>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                className="f16 fw9 pfc3 mb15"
              >
                Property Types in Bali
              </h3>,
              <div
                className="overflowXscroll swiper-full-width"
              >
                <div
                  className="flex mb8"
                >
                  <RenderLinks
                    name="Hotels in Bali"
                    url=""
                  />
                  <RenderLinks
                    name="Guest Houses in Bali"
                    url=""
                  />
                </div>
              </div>,
            ],
            "className": "p15 sbcw mb8",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Property Types in Bali",
                "className": "f16 fw9 pfc3 mb15",
              },
              "ref": null,
              "rendered": "Property Types in Bali",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="flex mb8"
                  >
                    <RenderLinks
                      name="Hotels in Bali"
                      url=""
                    />
                    <RenderLinks
                      name="Guest Houses in Bali"
                      url=""
                    />
                  </div>,
                ],
                "className": "overflowXscroll swiper-full-width",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <RenderLinks
                        name="Hotels in Bali"
                        url=""
                      />,
                      <RenderLinks
                        name="Guest Houses in Bali"
                        url=""
                      />,
                    ],
                    "className": "flex mb8",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "name": "Hotels in Bali",
                        "url": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "name": "Guest Houses in Bali",
                        "url": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
